plugins/woocommerce/src/StoreApi/Formatters/MoneyFormatter.php

```diff
@@ -23,12 +24,30 @@ public function format( $value, array $options = [] ) {
 			]
 		);
 
-		return (string) intval(
-			round(
-				( (float) wc_format_decimal( $value ) ) * ( 10 ** absint( $options['decimals'] ) ),
-				0,
-				absint( $options['rounding_mode'] )
-			)
-		);
+		// Ensure rounding mode is valid.
+		$rounding_modes           = [ PHP_ROUND_HALF_UP, PHP_ROUND_HALF_DOWN, PHP_ROUND_HALF_EVEN, PHP_ROUND_HALF_ODD ];
+		$options['rounding_mode'] = absint( $options['rounding_mode'] );
+		if ( ! in_array( $options['rounding_mode'], $rounding_modes, true ) ) {
+			$options['rounding_mode'] = PHP_ROUND_HALF_UP;
+		}
+
+		if ( ! is_int( $value ) && ! is_string( $value ) && ! is_float( $value ) ) {
+			wc_doing_it_wrong(
+				__FUNCTION__,
+				__( 'This function expects a $value arg of type INT, STRING or FLOAT.', 'woocommerce' ),
+				'9.2'
+			);
+
+			return '';
+		}
```

Comment:
Definitely not essential we change this, but might it make sense to move this higher (that is, perform this as our first check before doing any other work)?

I'm not suggesting this as a micro-optimization, because we're not really doing anything super expensive above this point, but it feels (to me) like a slightly more logical (and readable) way to order things.

Reply:
😅 indeed

done in https://github.com/woocommerce/woocommerce/pull/49361/commits/3767dedfcb114689066e4ac5107ddb806efa11e0